src/ert/gui/tools/plot/plot_api.py

```diff
@@ -157,7 +157,7 @@ def data_for_key(self, ensemble_name: str, key: str) -> pd.DataFrame:
         if key.startswith("LOG10_"):
             key = key[6:]
 
-        ensemble = self._get_ensemble(ensemble_name)
+        ensemble = self._get_ensemble_by_id(ensemble_name)
```

Comment:
Can you also change the function signature for `data_for_key`?

Reply:
`data_for_key` takes in the entire ensemble at the moment, and has 8 usages, so I think that could be a refactor for another PR as this is just targeted towards fixing the bug w/ overlapping plots.